waku/v2/waku_archive/driver/sqlite_driver/sqlite_driver.nim

```diff
@@ -59,22 +57,14 @@ proc new*(T: type SqliteDriver, db: SqliteDatabase): ArchiveDriverResult[T] =
   let insertStmt = db.prepareInsertMessageStmt()
   ok(SqliteDriver(db: db, insertStmt: insertStmt))
 
-method close*(s: SqliteDriver): ArchiveDriverResult[void] =
-  ## Close the database connection
-
-  # Dispose statements
-  s.insertStmt.dispose()
-
-  # Close connection
-  s.db.close()
-
-  ok()
-
-
-method put*(s: SqliteDriver, pubsubTopic: PubsubTopic, message: WakuMessage, digest: MessageDigest, receivedTime: Timestamp): ArchiveDriverResult[void] =
+method put*(s: SqliteDriver,
+            pubsubTopic: PubsubTopic,
+            message: WakuMessage,
+            digest: MessageDigest,
+            receivedTime: Timestamp):
+            Future[ArchiveDriverResult[void]] {.async.} =
   ## Inserts a message into the store
-
-  let res = s.insertStmt.exec((
+  return s.insertStmt.exec((
```

Comment:
I'd still assign this to a `res` variable here to keep the final `return` a bit cleaner.